### PR TITLE
Add job result download option to WebUI

### PR DESCRIPTION
**Pull request by markkuhn**
## Changes

This change allows WebUI users to download job results for local and s3 publisher types.

example

## Testing

Pushed long running jobs to EC2 fleet and verified that download button is only enabled if publisher is supported and job is not running. S3 and local jobs download successfully.

## Summary by CodeRabbit

- **New Features**
	- Added conditional rendering for "Publisher" information in the Job Information component.
	- Introduced a new DownloadButton for downloading job results based on publisher type and task status in the Job Executions component.
	- Enhanced Job Tabs to ensure Job Executions are only displayed when necessary job data is available.

- **Bug Fixes**
	- Improved robustness of Job Executions rendering by checking for required job data before display.

### CODE REVIEW

**Review by wdbaruni — CHANGES_REQUESTED** [review]
Outstanding work. Thanks for landing this in a short time!

I've tested the changes and the only issue I found is when a job have multiple results, such as the case when running multiple executions. We always end up downloading the result of the first execution. I don't suggest we aggregate anything in this case, but just display all the results to the user and let them decide what to download.

**Ways to re-produce this:**

```
# run devstack with three nodes and webui enabled
export PREDICTABLE_API_PORT=1;
bacalhau devstack

# in an another terminal, submit a job with count=3, which will ask for 3 executions
bacalhau docker run --publisher local --count 3 hello-world

# downloading from the webui will only download the first result 
```

Here is the API output for reference. Don't worry about pagination for now as it doesn't actually work for this API and we just return all results, I think!
```
→ curl 127.0.0.1:1234/api/v1/orchestrator/jobs/j-05a1e060-8a4c-41b8-a23e-b0b210cfdf8d/results | jq
{
  "NextToken": "",
  "Items": [
    {
      "Type": "urlDownload",
      "Params": {
        "URL": "http://127.0.0.1:51837/e-29f22fb1-bf5e-4f36-9391-19715b9d0ab1.tar.gz"
      }
    },
    {
      "Type": "urlDownload",
      "Params": {
        "URL": "http://127.0.0.1:51841/e-b85ff14d-ec5c-4954-9f22-112e32648a67.tar.gz"
      }
    },
    {
      "Type": "urlDownload",
      "Params": {
        "URL": "http://127.0.0.1:51844/e-eef8a55b-b8a8-4558-971d-b7d7f45792c0.tar.gz"
      }
    }
  ]
}
```

**Suggestions**:
1. Current the current download button, and show a popup with list of results to download iff there are multiple results
2. Have a separate tab for results and show list of available results

File: webui/components/jobs/details/JobExecutions.tsx [hunk 1]
```diff
@@ -8,57 +7,248 @@ import {
...
+                  )}
+                </TableCell>
+                <TableCell>
+                  <DownloadButton
+                    publisherType={publisherType}
+                    publishedResult={execution.PublishedResult}
+                    isTaskCompleted={isTaskCompleted}
+                    onDownload={onDownload}
+                  />
+                </TableCell>
```
[the remark this answers is not in the document]
Comment (markkuhn) [reply webui/components/jobs/details/JobExecutions.tsx:244]: From what I understand a publisher is defined at the task level so this is not applicable.

File: webui/components/jobs/details/JobExecutions.tsx [hunk 2]
```diff
@@ -8,57 +7,248 @@ import {
...
+  executions?: apimodels_ListJobExecutionsResponse,
+  tasks: Array<models_Task>,
+  state: models_State_models_JobStateType
+}) => {
+  const jobState = getJobState(state?.StateType);
+  const isTaskCompleted = jobState ? COMPLETED_JOB_STATES.includes(jobState) : false;
+  const publisherType = useMemo(() =>
+    tasks[0]?.Publisher?.Type ?? "UNKNOWN",
+    [tasks]
+  );
```
[the remark this answers is not in the document]
Comment (markkuhn) [reply webui/components/jobs/details/JobExecutions.tsx:191]: Since task is an Array, if it is empty then "UNKNOWN" will be returned.

**Review by wdbaruni — APPROVED** [review]
Nice work!

**Pull request merged**